File: sdk-objectivec/office365_odata_base/office365_odata_impl/Implementations/MSOAuthCredentials.m
//
//  MSAuthCredentials.m
//  office365_odata_impl
//
//  Created by Gustavo on 10/15/14.
//  Copyright (c) 2014 Microsoft. All rights reserved.
//

#import "MSOAuthCredentials.h"

@interface MSOAuthCredentials()

@property NSString* token;

@end

@implementation MSOAuthCredentials

-(void)addToken : (NSString*) token{
    self.token = token;
}

-(void)prepareRequest:(id<MSRequest>)request{
    [request addHeader:@"Authorization" :[[NSString alloc] initWithFormat:@"Bearer %@", self.token]];
}

@end

File: sdk-swift/office365_odata_impl/MSOAuthCredentials.swift
/*******************************************************************************
* Copyright (c) Microsoft Open Technologies, Inc.
* All Rights Reserved
* Licensed under the Apache License, Version 2.0.
* See License.txt in the project root for license information.
******************************************************************************/

import Foundation

public class MSOAuthCredentials : MSCredentials{

    private var token : String;
    
    public init(token : String){
        self.token = token;
    }
    
    public func prepareRequest(request : MSRequest){
        request.addHeader("Authorization", value: "Bearer " + self.token)
    }
}